downloader/artifact_downloader.go

```diff
@@ -283,6 +295,15 @@ func (ad *ConcurrentArtifactDownloader) createClient() *retryablehttp.Client {
 
 		return shouldRetry, err
 	}
+	client.ErrorHandler = func(resp *http.Response, err error, numTries int) (*http.Response, error) {
+		statusCode := -1
+		if resp != nil {
+			statusCode = resp.StatusCode
+		}
+		ad.Logger.Warnf("After %d retries http status: %d, error: %s", numTries, statusCode, err)
+
+		return resp, err
+	}
```

Comment:
This error handler is called at the end after all of the retries were performed and there is still an error. The function receives how many retries there were which I am logging out.